crates/katana/runner/src/lib.rs

```diff
@@ -188,6 +208,109 @@ fn determine_default_program_path() -> String {
     if let Ok(bin) = std::env::var("KATANA_RUNNER_BIN") { bin } else { "katana".to_string() }
 }
 
+type Provider = FillProvider<
+    JoinFill<
+        JoinFill<JoinFill<JoinFill<Identity, GasFiller>, NonceFiller>, ChainIdFiller>,
+        SignerFiller<EthereumSigner>,
+    >,
+    RootProvider<Http<Client>>,
+    Http<Client>,
+    Ethereum,
+>;
+
+#[derive(Debug)]
+pub struct AnvilRunner {
+    process: Child,
+    provider: Provider,
+    pub endpoint: String,
+    address: Address,
+    secret_key: String,
+}
```

Comment:
as per my comment on the previous PR https://github.com/dojoengine/dojo/pull/1870#discussion_r1587338563, dont think that we need to define the runner ourselves now that `alloy` seems to have an `Anvil` binding here:

https://github.com/alloy-rs/alloy/blob/main/crates/node-bindings/src/anvil.rs

cc @glihm

Reply:
Went through alloy repo, and yes @fabrobles92 as @kariy is pointing out, alloy has an anvil runner implementation. Would you mind updating to this in order to rely on their implementation and support? :)